java/src/main/java/io/cucumber/gherkin/Token.java

```diff
@@ -2,24 +2,26 @@
 
 import io.cucumber.messages.types.Location;
 import io.cucumber.messages.types.StepKeywordType;
+import org.jspecify.annotations.NonNull;
+import org.jspecify.annotations.Nullable;
 
 import java.util.List;
 
 import static java.util.Objects.requireNonNull;
 
 final class Token {
-    final Line line;
+    final @Nullable Line line;
     final boolean eof;
-    Parser.TokenType matchedType;
-    String matchedKeyword;
-    String matchedText;
-    List<LineSpan> matchedItems;
+    Parser.@Nullable TokenType matchedType;
+    @Nullable String matchedKeyword;
+    @Nullable String matchedText;
+    @Nullable List<LineSpan> matchedItems;
     int matchedIndent;
-    String matchedLanguage;
-    StepKeywordType keywordType;
+    @Nullable String matchedLanguage;
+    @Nullable StepKeywordType keywordType;
     Location location;
 
-    private Token(Line line, Location location) {
+    private Token(@Nullable Line line, Location location) {
```

Comment:
Ideally we have two different types of token here. Can we use records?

Reply:
I don't think the EOF token is needed. If we have a careful look at `TokenScanner.read` we can probably replace it with `null`.